### ISSUE TITLE
Support issues without associated files

### ISSUE THREAD

**Issue by Phyks** [wenns/sonar-cxx#468]
Hi,

[PC-Lint](http://www.gimpel.com/html/index.htm) is generating issues without any associated filename for some issues (typically when two threads access the same object without precautions). In this case, SonarQube displays the issue in the interface, with the associated number of occurrence but that is all.

The problem is that the rule is general, with "Symbol" in it, not being replaced in the global issues drilldown. As there is no further drilldown for such issues (because there is no filename associated), one cannot see the detailed errors and has to look at the pclint report itself.

Maybe it could be possible to handle such issues without filenames as separate issues, global for the project, and not being part of the same rule?

See also issues https://github.com/wenns/sonar-cxx/issues/116 and https://github.com/wenns/sonar-cxx/pull/81.

**Comment by guwirth** [comment]
@Phyks Can you provide an example please?

**Comment by Phyks** [comment]
@guwirth: I am having troubles with the issue number 457 on [this PC-Lint manual](http://www.gimpel.com/html/manual.pdf) (page 72). 

The rule is:
```
Thread ’Symbol1’ has an unprotected write access to variable ’Symbol2’ which is used by thread ’Symbol3’
```

Symbols are declared in different files. Then, in the PC-Lint report, there is no filename associated to this rule and in SonarQube, I have the rule listed as an issue, with the number of occurrence, but as a generic rule (meaning that Symbol1, Symbol2 and Symbol3 are not replaced). I cannot get a drilldown for this rule, to list all the matching symbols in the interface.


I can provide a more detailed example tomorrow (GMT) if needed.

**Comment by Phyks** [comment]
Here is the relevant line from PC-Lint XML report that I feed into SonarQube:
```
<issue file ="" line = "0" number = "457" desc = "Function 'classA::threadA(void)' of thread 'classA::startThreadA(void *)' has an unprotected write access to variable 'nStep_' which is used by function 'classA::threadB(void)' of thread 'classA::startThreadB(void *)'"/>
```

As you can see, there is no associated file and it seems PC-Lint cannot set any filename associated to such issues.

**Comment by guwirth** [comment]
@Phyks: Could you add an example with at least two such issues please?

**Comment by Phyks** [comment]
Here it is:

```
<issue file ="" line = "0" number = "457" desc = "Function 'classA::threadA(void)' of thread 'classA::startThreadA(void *)' has an unprotected write access to variable 'nStep_' which is used by function 'classA::ThreadA(void)' of thread 'classA::startThreadA(void *)'"/>
<issue file ="" line = "0" number = "457" desc = "Function 'Logger::write()' of thread 'classA::startThreadA(void *)' has an unprotected write access to variable '_Stderr' which is used by function 'logger::Logger::write(const char *, enum logger::logginglevel::Enum)' of thread 'classA::startThreadA(void *)'"/>
```

Thanks!

**Comment by guwirth** [comment]
@Phyks just a short feedback without looking into the details:
1) Issues without or an empty file tag are project issues, they are assigned to the project and not a file.
2) The cxx plugin has a issue filter for duplicated issues. If file+line+number+desc is equal to another issue it is not added a second time. 
3) The message you see in the overview is always a general one from the rule file (see https://github.com/wenns/sonar-cxx/wiki/Extending-the-code-analysis). Only if you are jumping into the file you see the concrete message without the placeholders (which is currently not possible in your case). 

Think this is a SQ core functionality. Maybe you can ask in parallel in the SQ forum how they deal with project issues?

**Comment by Phyks** [comment]
Ok, I'll post on SQ core as well :)

**Comment by Bertk** [comment]
Did you query for issues of rule L0457? There are some UI improvements with SQ5.1 see
http://www.sonarqube.org/sonarqube-5-1-in-screenshots/
![capture](https://cloud.githubusercontent.com/assets/2315215/7428063/85cb595e-efe9-11e4-8a7a-668c39871eaa.JPG)

I can query for PC-lint issues which have no file name e.g. L0974.

**Comment by Phyks** [comment]
I can confirm this is resolved in SQ 5.1. Thanks!
